pkg/util/log/audit/audit.go

```diff
@@ -65,125 +72,138 @@ var (
 
 // NewEntry returns a log entry that embeds the provided logger. It has a hook
 // that knows how to hydrate an IFxAudit log payload before logging it.
-func NewEntry(env env.Core, logger *logrus.Logger) *logrus.Entry {
+func NewEntry(logger *logrus.Logger) *logrus.Entry {
 	logger.AddHook(&payloadHook{
-		payload: &AuditPayload{},
-		env:     env,
+		payload: &payload{},
 	})
-
 	return logrus.NewEntry(logger)
 }
 
 // payloadHook, when fires, hydrates an IFxAudit log payload using data in a log
 // entry.
 type payloadHook struct {
-	payload *AuditPayload
-	env     env.Core
+	payload *payload
 }
 
 func (payloadHook) Levels() []logrus.Level {
 	return logrus.AllLevels
 }
 
 func (h *payloadHook) Fire(entry *logrus.Entry) error {
-	payload := *h.payload // shallow copy
+	h.payload = &payload{}
```

Comment:
I'm not super familiar with hooks. When constructing a logger emitter is this object is shared in memory or not?
When reusing the same memory (previously you were doing a copy of it). I always wonder will we have 2 requests dealing with the same memory and modifying the same object?

Reply:
AFAICT, the hooks are [thread-safe](https://github.com/sirupsen/logrus/blob/cd4bf4ef8de16b243cce0e062742feb34128648b/entry.go#L257-L264). Thanks for bringing this up - I had the same suspicion too.